sbin/build.sh

```diff
@@ -253,6 +253,169 @@ patchFreetypeWindows() {
   fi
 }
 
+# This attempts to compare the jdk version from version-numbers.conf with arg 1.
+# We will then return whichever version is bigger/later.
+# e.g. 17.0.1+32 > 17.0.0+64
+# If any errors or unusual circumstances are detected, we simply return arg1 to avoid destabilising the build.
+compareToOpenJDKFileVersion(){
+  # First, sanity checking on the arg.
+  if [ $# -eq 0 ]; then
+    echo "compareToOpenJDKFileVersion_was_called_with_no_args"
+    exit 1
+  fi
+  if [[ ! "$1" =~ ^jdk\-[0-9]+[0-9\.]*\+[0-9]+$ ]]; then
+    if [[ ! "$1" =~ ^8u[0-9]+\-b[0-9][0-9]$ ]]; then
+      echo "$1"
+      exit 1
+    fi
+  fi
+
+  build_src="${BUILD_CONFIG[WORKSPACE_DIR]}/${BUILD_CONFIG[WORKING_DIR]}/${BUILD_CONFIG[OPENJDK_SOURCE_DIR]}"
+  fileVersionArray=("0" "0" "0" "0" "1")
+  error="false"
+  numbersFile=""
+
+  # Parse the version-numbers.conf or the equivalent.
+  if [ "${BUILD_CONFIG[OPENJDK_FEATURE_NUMBER]}" -gt 8 ]; then
+    numbersFile="${build_src}/make/conf/version-numbers.conf"
+    if [ "${BUILD_CONFIG[OPENJDK_FEATURE_NUMBER]}" -eq 11 ]; then
+      numbersFile="${build_src}/make/autoconf/version-numbers"
+    fi
+
+    # Check the file exists.
+    if [ ! -r "${numbersFile}" ]; then
+      echo "$1"
+      exit 1
+    fi
+
+    # File parsing logic for jdk11+
+    fileVersionArray[0]=$(grep "DEFAULT_VERSION_FEATURE" "${numbersFile}" | head -1 | grep -Eo '[0-9]+') || error="true"
+    fileVersionArray[1]=$(grep "DEFAULT_VERSION_INTERIM" "${numbersFile}" | head -1 | grep -Eo '[0-9]+') || error="true"
+    fileVersionArray[2]=$(grep "DEFAULT_VERSION_UPDATE" "${numbersFile}" | head -1 | grep -Eo '[0-9]+') || error="true"
+    fileVersionArray[3]=$(grep "DEFAULT_VERSION_PATCH" "${numbersFile}" | head -1 | grep -Eo '[0-9]+') || error="true"
+    fileVersionArray[4]="1"
```

Comment:
This is the temurin patch number?

Reply:
No, this is the default build number. We can't set it to "0", because then configure treats it as us building without a build number.